### PR TITLE
Add common widget for vector3

### PR DESCRIPTION
**Pull request by AzulRadio**
Signed-off-by: youhy <haoyuan2019@outlook.com>

* Part of https://github.com/gazebosim/gz-gui/issues/310

# New feature

## Summary
Add common widget for vector3

## Test it
Use the following test QML to test: 
```
import QtQuick 2.9
import QtQuick.Layouts 1.3

GzVector3 {
  id: gzVector
  property double xModel: 120
  property double yModel: 110
  property double zModel: 100
  xName: "Red"
  yName: "Green"
  zName: "Blue"
  xSpin.value: xModel
  ySpin.value: yModel
  zSpin.value: zModel
  onGzVectorSet: {
    xModel = xSpin.value
    yModel = ySpin.value
    zModel = zSpin.value
    console.log(xModel, yModel, zModel)
  }
}
```
Then use
```
qmlscene Test.qml
```
to test
## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/ignition/gui/qml/GzVector3.qml [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  height: gzVectorContent.height
+
+  // Left indentation
+  property int indentation: 10
+
+  // Horizontal margins
+  property int margin: 5
+
+  // Maximum spinbox value
+  property double spinMax: 1000000
```
Comment (chapulina) [inline include/ignition/gui/qml/GzVector3.qml:87]:
```suggestion
  property double spinMax: Number.MAX_VALUE
```

File: include/ignition/gui/qml/GzVector3.qml [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  property string zName: "Z"
+
+  // Units, defaults to meters.
+  // Set to "" to omit the parentheses.
+  property string gzUnit: "m"
+
+
+  /*** The following are private variables: ***/
+  // Show Pose bar (used to control expand)
+  property bool show: true
```
Comment (chapulina) [inline include/ignition/gui/qml/GzVector3.qml:76]: Let's make it consistent with pose, call it expand and make it public

File: include/ignition/gui/qml/GzVector3.qml [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+  height: gzVectorContent.height
+
+  // Left indentation
+  property int indentation: 10
+
+  // Horizontal margins
+  property int margin: 5
```
Comment (chapulina) [inline include/ignition/gui/qml/GzVector3.qml:84]: Let's move the marging and indentation logic to the parent of this widget

File: include/ignition/gui/qml/GzVector3.qml [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+  // Maximum spinbox value
+  property double spinMax: 1000000
+
+  // local variables to store spinbox values
+  property var xItem: {}
+  property var yItem: {}
+  property var zItem: {}
+
+  // Dummy component to use its functions.
+  IgnHelpers {
```
Comment (chapulina) [inline include/ignition/gui/qml/GzVector3.qml:95]: You could use the one from `IgnHelpers.qml` here

File: include/ignition/gui/qml/GzVector3.qml [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+ *    readOnly: false
+ *    xValue: xValueFromCPP
+ *    yValue: yValueFromCPP
+ *    zValue: zValueFromCPP
+ *    onGzVectorSet: {
+ *      myFunc(_x, _y, _z, _roll, _pitch, _yaw)
+ *    }
+ *  }
+**/
+Rectangle {
```
Comment (chapulina) [inline include/ignition/gui/qml/GzVector3.qml:44]: The widget's background color is plain white, but it should follow the parent's color.

Before this PR:

image


With this PR:

image

**Review by chapulina — CHANGES_REQUESTED** [review]

File: include/ignition/gui/qml/GzVector3.qml [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+   */
+  signal gzVectorSet(double _x, double _y, double _z)
+
+  // Names for XYZ
+  property string xName: "X"
+  property string yName: "Y"
+  property string zName: "Z"
+
+  // Units, defaults to meters.
+  // Set to "" to omit the parentheses.
```
Comment (jennuine) [inline include/ignition/gui/qml/GzVector3.qml:70]:
```suggestion
  // Set to "" to omit the units & parentheses.
```

File: include/ignition/gui/qml/GzVector3.qml [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+  // Expand/Collapse of this widget
+  property bool expand: true
+
+
+  /*** The following are private variables: ***/
+  height: gzVectorContent.height
+
+  // Maximum spinbox value
+  property double spinMax: Number.MAX_VALUE
```
Comment (jennuine) [inline include/ignition/gui/qml/GzVector3.qml:81]: This should be moved out of the private variables section since users can overwrite

**Review by jennuine — APPROVED** [review]

**Pull request merged**